src/gg.regression.unity.bots/Runtime/Scripts/StateRecorder/ProfilerObserver.cs

```diff
@@ -0,0 +1,103 @@
+﻿using System.Collections.Generic;
+using Unity.Profiling;
+using UnityEngine;
+
+namespace StateRecorder
+{
+    public struct ProfilerObserverResult
+    {
+        // Amount of memory (in bytes) the operating system reports in use by the application
+        public long? systemUsedMemory;
+
+        // Used heap size (in bytes) that is garbage collected
+        public long? gcUsedMemory;
+
+        // Time spent (in milliseconds) by the CPU on the main thread since the last tick
```

Comment:
One remaining question/concern, sorry I didn't notice this the first time

This says `in milliseconds` but is a `double`.  Is this supposed to be an integer type (ie: `long`) ? or is this time actually measured in some other unit ?

Reply:
This was something I changed as part of this PR. The profiler outputs in `long` nanoseconds, but I found that for large frame counts summing everything up would cause an overflow in the JSON output (even after converting it to double for some reason). So, I now output in milliseconds double to avoid this problem.

Reply:
Hmm, this doesn't seem right.  Lets figure out why you were seeing overflow because we should be able to add those longs without any real issue since they aren't that large.

Max c# signed `long` is
9,223,372,036,854,775,807

A whole day in nanoseconds is
86,400,000,000,000

Reply:
Ok so, the following code gives me "-2147483648" as output:
`DoubleJsonConverter.WriteToStringBuilderNullable(sb, 86_400_000_000_000.0);`

So, maybe there's a bug there?

Using `LongJsonConverter` works fine though, so I suppose given your example it could make sense to continue using longs for summing up the time (I was basically just following the Unity docs approach to summing up the profiler outputs as shown here: https://docs.unity3d.com/ScriptReference/Unity.Profiling.ProfilerRecorder.html)

Reply:
Yeah, lets go with longs for tracking/summing it up so we don't overflow or have weird FP behavior or format

Reply:
Updated to use long nanoseconds again.